src/varcache.c

```diff
@@ -99,12 +180,17 @@ static int apply_var(PktBuf *pkt, const char *key,
 	if (strcasecmp(cval->str, sval->str) == 0)
 		return 0;
 
-	/* the string may have been taken from startup pkt */
-	if (!pg_quote_literal(qbuf, cval->str, sizeof(qbuf)))
+	if (strcasecmp("search_path", key) == 0) {
+		tmp = cval->str;
+	}
+	else if (pg_quote_literal(qbuf, cval->str, sizeof(qbuf))){
+		tmp = qbuf;
+	}
+	else
 		return 0;
 
 	/* add SET statement to packet */
-	len = snprintf(buf, sizeof(buf), "SET %s=%s;", key, qbuf);
+	len = snprintf(buf, sizeof(buf), "SET %s=%s;", key, tmp);
```

Comment:
If it doesn't fit, let's use a allocated buffer instead of logging a warning that the value is too long (line 199)